account_invoice_change_currency/tests/test_account_invoice_change_currency.py

```diff
@@ -81,24 +81,22 @@ def test_change_invoice_currency(self):
         before_curr = inv.currency_id
         before_amount = inv.amount_total
         after_curr = self.env.ref('base.USD')
-        wiz = self.env['wizard.change.invoice.currency'].\
-            with_context(active_id=inv.id).create(
-                {'currency_id': after_curr.id})
-        wiz.button_change_currency()
-        expected_value = before_curr.with_context(date=fields.Date.today()).\
-            compute(before_amount, after_curr)
+        inv.write({'currency_id': after_curr.id})
+        inv.action_account_change_currency()
+        expected_value = before_curr._convert(
+            before_amount, after_curr, inv.company_id, fields.Date.today())
 
         self.assertEqual(
             inv.amount_total, expected_value,
             'Total amount of invoice does not equal to expected value!!!')
 
     def test_change_validated_invoice_currency(self):
         inv = self.create_simple_invoice(fields.Date.today())
-        wiz = self.env['wizard.change.invoice.currency'].\
-            with_context(active_id=inv.id).create(
-                {'currency_id': self.env.ref('base.USD').id})
-        # Validate invoice before change
+        before_amount = inv.amount_total
         inv.action_invoice_open()
         # Make sure that we can not change the currency after validated:
-        with self.assertRaises(exceptions.UserError):
-            wiz.button_change_currency()
+        inv.write({'currency_id': self.env.ref('base.USD').id})
+        inv.action_account_change_currency()
+        self.assertEqual(
+            inv.amount_total, before_amount,
+            'Total amount of invoice does not equal to expected value!!!')
```

Comment:
Use `self.uid = a_new_billing_user` in the `setUp` method of the tests assigned to a new user with custom permissions available to press the button.
In order to avoid a regression error for a permission error:
 - ![screen shot 2018-11-06 at 20 44 04](https://user-images.githubusercontent.com/6644187/48107466-64e14980-e205-11e8-8e12-d663810c3ada.png)